### PR TITLE
Add basic junit formatter

### PR DESCRIPTION
**Pull request by jkrentz**
exposing the commonjs module under formatters/

### CODE REVIEW

File: src/junitFormatter.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        var xml = [];
+
+        xml.push("<?xml version=\"1.0\" encoding=\"utf-8\"?>");
+        if (failures.length > 0) {
+            xml.push("<testsuites>");
+            xml.push(this.testsuiteStartXML(failures));
+            xml = xml.concat(failures.map(failure => this.testcaseXML(failure)));
+            xml.push("</testsuite>");
+        }
+        xml.push("</testsuites>");
```
Comment (nickknw) [inline src/junitFormatter.ts:65]: Should this closing `</testsuites>` go inside the if statement as well?

File: formatters/junitFormatter.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+"use strict";
```
Comment (nickknw) [inline formatters/junitFormatter.js:1]: Does the generated js source need to be in here?

**Comment by nickknw** [comment]
Awesome! LGTM

File: formatters/junitFormatter.js (hunk 2 above)
Comment (jkrentz) [reply formatters/junitFormatter.js:1]: So tslint needs cjs file as input. When I tried to pass it ts it balked at me.

I think the magic I'm missing is a prepublish task in the package.json https://docs.npmjs.com/misc/scripts#common-uses

So the cjs wouldn't be in source control but would be in npm.

File: src/junitFormatter.ts (hunk 1 above)
Comment (jkrentz) [reply src/junitFormatter.ts:65]: I think the opening tag should be outside the if block. Thanks.

**Pull request merged**